pt-liferay-solutions/liferay-portal
# [LPS-146474 & LPS-146475 & LPS-146478 & LPS-146495] [CP 2.0] DXP Activation Keys Table Management

### How it was developed?
It was developed a table with pagination and checkbox, including a status bar filter.

We've made some important changes in a few components

- Changed the way generic Table works (2fb780bf3d4eeba272fbe619c7501eed3c57c831)
- Changed the logic in TablePagination (2fb780bf3d4eeba272fbe619c7501eed3c57c831)
- Added a generic GroupButton component (2ed2b644d59458c766cfdb69b0b36a28dc1921ea)

![image](https://user-images.githubusercontent.com/44975114/151984830-3df54ed5-a1a3-43f1-9c6d-3c76b1d63135.png)

**Table component**:

- Now all the pagination configuration will be handled by the **pagination config prop**, eg:
`paginationConfig={{
						activePage,
						itemsPerPage,
						labels: {
							paginationResults: 'Showing {0} to {1} of {2}',
							perPageItems: 'Show {0} Items',
							selectPerPageItems: '{0} Items',
						},
						listItemsPerPage: [
							{label: 5},
							{label: 10},
							{label: 20},
							{label: 50},
						],
						setActivePage,
						setItemsPerPage,
						showDeltasDropDown: true,
						totalCount,
					}}`
- New props added in Table Pagination

1. labels -> Pass this prop if you want to customize the return of the DropDown itemsPerPage in pagination
2. listItemsPerPage -> An array that contains how many options will be shown in itemsPerPage
3. showDeltasDropDown -> Set this prop to true in order to enable the dropDown itemsPerPage selection

- New props added to the generic table:

1.    hasCheckbox -> Pass this prop if you need checkbox in the table

- New props added to Table Row (Column):

1. columns.header.noWrap -> Pass this prop to enforce the column header to not break in line (https://github.com/pt-liferay-solutions/liferay-portal/pull/790/commits/638c43df741e085a99315aec85477e17cfd14069#diff-9668d6cd612d36d1f016eb98eee20cc2769ccae24aeea28110f391ed60804d97R132)

### What has been affected by the development ?

- As we created the GroupButton component, We've enforced the component in:
 ([2ed2b644d59458c766cfdb69b0b36a28dc1921ea](https://github.com/pt-liferay-solutions/liferay-portal/commit/2ed2b644d59458c766cfdb69b0b36a28dc1921ea#diff-311ffd8c953fe34a3bcbb016b58b2e4400cce17267a0b82a8f65277757403d7c)) (Dependent in LPS-140163)
- As we made some big changes in the Table component styles & logic, we're affecting directly the 
(LPS-140164 & LPS-140183)

### Adding data mock up to test

1. Go to Control Panel -> Account Users
2. Search for the user you're using
3. In Accounts -> Select a new Account
4. Search for 3809080 Account (2022 FIFA World Cup Qatar - FAN ID) and add to the user
4. Open the Customer Portal site and the account you've selected
5. In sidebar -> Product Activation -> DXP. 
6. You should be able to see the table

### Test Plan

1. Click on one status bar filter option and see that the table should adjust by the filter.
2. Try changing pagination / items per page drop down to change the table visualization
3. Hover your mouse on Status column row by row, see that it should appear a tool tip showing the status by color.
4. Hover your mouse on Environment Name column. It should appear a tool tip showing the name and description of the activation key
5. When resizing the window, the Environment Name column should truncate the text and not break the line
6. If Key Type = Virtual Cloud, it should appear the Cloud image and detail how many cluster nodes it has
7. In Exp. Date column, if we have + 100 years from now it should appear DNE field. When hovering it, should appear a tool tip showing "This key does not expire"
8. Click on download button at the end of the row, this should download the file


PS: Make sure that you're authenticated with Okta sign in. (User Garrett)
